### ISSUE TITLE
[v7r1] JobMonitoring (ElasticSearchDB) keeps opening files

### ISSUE THREAD

**Issue by andresailer** [DIRACGrid/DIRAC#4770]
The JobMonitoring service keeps opening files until "too many open files"

```
# for pid in [0-9]*; do     echo "$(ls /proc/$pid/fd/ | wc -l) file descriptors for $pid"; done | sort -n | tail -n1
1023 file descriptors for 27485
# ps -q 27485 aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
dirac    27485 15.0  1.9 1061288 144248 ?      Sl   15:00  28:46 python /opt/dirac/pro/DIRAC/Core/scripts/dirac-service.py WorkloadManagement/JobMonitoring --cfg /opt/dirac/pro/etc/WorkloadManagement_JobMonitoring.cfg
```
Thankfully there is an exception, which tells us the culprit. There are also thousands of "/tmp/cas.*" files on the certification machine.

```
2020-10-19 15:57:14 UTC WorkloadManagement/JobMonitoring ERROR: Server error while loading handler: [Errno 24] Too many open files: '/tmp/cas.cjkZe3.pem'
Traceback (most recent call last):
File "/opt/dirac/pro/DIRAC/Core/DISET/private/Service.py", line 536, in _instantiateHandler
handlerInstance.initialize()
File "/opt/dirac/pro/DIRAC/WorkloadManagementSystem/Service/JobMonitoringHandler.py", line 70, in initialize
gElasticJobParametersDB = ElasticJobParametersDB()
File "/opt/dirac/pro/DIRAC/WorkloadManagementSystem/DB/ElasticJobParametersDB.py", line 70, in __init__
super(ElasticJobParametersDB, self).__init__(name, 'WorkloadManagement/ElasticJobParametersDB', indexPrefix)
File "/opt/dirac/pro/DIRAC/Core/Base/ElasticDB.py", line 45, in __init__
useSSL=self.__useSSL)
File "/opt/dirac/pro/DIRAC/Core/Utilities/ElasticSearchDB.py", line 95, in __init__
retVal = bd.getCAs()
File "/opt/dirac/pro/DIRAC/FrameworkSystem/Client/BundleDeliveryClient.py", line 162, in getCAs
retVal = Utilities.generateCAFile()
File "/opt/dirac/pro/DIRAC/Core/Security/Utilities.py", line 61, in generateCAFile
fn = tempfile.mkstemp(prefix="cas.", suffix=".pem")[1]
File "/opt/dirac/pro/diracos/usr/lib64/python2.7/tempfile.py", line 314, in mkstemp
File "/opt/dirac/pro/diracos/usr/lib64/python2.7/tempfile.py", line 244, in _mkstemp_inner
OSError: [Errno 24] Too many open files: '/tmp/cas.cjkZe3.pem'
```

Though I am not sure if the problem is that for each JobMonitoring call the ElasticSearchDB instance is created, instead of using the global instance, or if the CAs file should be closed, or not created in the first place, or some config issue on the server prevents that.

**Comment by fstagni** [comment]
I don't have time to check now but I most probably introduced a bug in https://github.com/DIRACGrid/DIRAC/pull/4747

**Comment by andresailer** [comment]
Ahh, yes, probably this
https://github.com/DIRACGrid/DIRAC/pull/4747/files#r507890269

**Comment by fstagni** [comment]
I think fixed in #4767
